include/flatbuffers/flatbuffers.h

```diff
@@ -599,6 +599,21 @@ class vector_downward {
     scratch_ = buf_;
   }
 
+  // Relinquish the pointer to the caller.
+  uint8_t *release_raw(size_t &size, size_t &offset) {
+    uint8_t *buf = buf_;
+    size = reserved_;
+    offset = static_cast<size_t>(cur_ - buf_);
+
+    if (own_allocator_ && allocator_) { delete allocator_; }
```

Comment:
maybe put these 3 lines in a `clear_allocator` and also call it from the destructor?